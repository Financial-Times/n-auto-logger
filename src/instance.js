import nLogger from '@financial-times/n-logger';

let logger = nLogger;

export const setupLoggerInstance = instance => {
	logger = instance;
};

export const getLoggerInstance = () => logger;
