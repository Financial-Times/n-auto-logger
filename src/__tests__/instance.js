import nLogger from '@financial-times/n-logger';

import { setupLoggerInstance, getLoggerInstance } from '../instance';
import createEventLogger from '../event-logger';

describe('instance', () => {
	it('default to n-logger', () => {
		const logger = getLoggerInstance();
		expect(logger).toBe(nLogger);
	});

	it('can be set to other instance and log properly', async () => {
		const alternative = {
			info: jest.fn(),
			warn: jest.fn(),
			error: jest.fn(),
		};
		setupLoggerInstance(alternative);
		const logger = getLoggerInstance();
		expect(logger).toBe(alternative);
		const event = createEventLogger({ event: 'USER-SIGNUP' });
		await event.start();
		await event.success();
		await event.failure({ message: 'user with the email already exists' });
		expect(alternative.info.mock.calls).toMatchSnapshot();
		expect(alternative.warn.mock.calls).toMatchSnapshot();
		expect(alternative.error.mock.calls).toMatchSnapshot();
	});
});
