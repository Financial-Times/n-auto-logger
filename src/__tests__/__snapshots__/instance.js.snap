// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instance can be set to other instance and log properly 1`] = `
Array [
  Array [
    Object {
      "event": "USER-SIGNUP",
    },
  ],
  Array [
    Object {
      "event": "USER-SIGNUP",
      "result": "success",
    },
  ],
]
`;

exports[`instance can be set to other instance and log properly 2`] = `Array []`;

exports[`instance can be set to other instance and log properly 3`] = `
Array [
  Array [
    Object {
      "category": "CUSTOM_ERROR",
      "event": "USER-SIGNUP",
      "message": "user with the email already exists",
      "result": "failure",
    },
  ],
]
`;
